### PR TITLE
Add different fixes for gcc 7 warnings.

### PR DESCRIPTION
**Pull request by 4144**
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1450)

### CODE REVIEW

File: src/map/skill.c [hunk 1]
```diff
@@ -7630,8 +7643,11 @@ int skill_castend_nodamage_id(struct block_list *src, struct block_list *bl, uin
 				sc_start4(src,bl,type,100,skill_lv,skill_id,src->id,skill->get_time(skill_id,skill_lv),1000));
 		#ifndef RENEWAL
 			if (sd) skill->blockpc_start (sd, skill_id, skill->get_time(skill_id, skill_lv)+3000);
-		#endif
+				break;
+			__attribute__ ((fallthrough));
```
Comment (dastgir) [inline src/map/skill.c]: fallthrough after break?

Comment (4144) [inline src/map/skill.c]: break here inside if
This mean code may ignore break

**Comment by hemagx** [comment]
can you please include the compile log?

File: src/map/skill.c (hunk 1 above)
Comment (dastgir) [inline src/map/skill.c]:
```
if (sd) skill->blockpc_start (sd, skill_id, skill->get_time(skill_id, skill_lv)+3000);
```
if only includes skill->blockpc_start

**Pull request merged**